### PR TITLE
Update to Babel 6

### PR DESCRIPTION
**Pull request by ockham**
While we're at it. cc @blowery @rralian

### CODE REVIEW

**Comment by ockham** [comment]
`npm install` and `npm test` work locally, but apparently not on CircleCI :-/ @aduth since you did the Calypso Babel 6 upgrade -- is this an error you're familiar with?

File: package.json [hunk 1]
```diff
@@ -16,15 +16,17 @@
     "node": ">=4.2.3"
   },
   "scripts": {
-	"compile": "babel -sd lib/ src/",
-	"prepublish": "npm run compile",
-	"test": "make test"
+    "compile": "babel -sd lib/ src/",
+    "prepublish": "npm run compile",
+    "test": "mocha --require babel-core/register --reporter spec test/test.jsx"
```
Comment (aduth) [inline package.json]: I think you want to be using the [`babel-register`](https://babeljs.io/docs/usage/require/) package, not `babel-core/register`.

**Comment by aduth** [comment]
Aside: Noted that CircleCI runs under npm 2.x . We should probably consider upgrading (I'd guess driven off `engines` in `package.json` ?) (related: [T2706](https://phabricator.babeljs.io/T2706))

**Comment by aduth** [comment]
@ockham Restarted the build without cache and looks to be passing this time around

File: package.json (hunk 1 above)
Comment (ockham) [inline package.json]: > I think you want to be using the [`babel-register`](https://babeljs.io/docs/usage/require/) package, not `babel-core/register`.

b3947c9

**Comment by ockham** [comment]
> @ockham Restarted the build without cache and looks to be passing this time around

Hah, thanks a lot! I did restart the build but apparently missed the option to discard the cache.

Good to merge?

**Comment by aduth** [comment]
Took it for a spin and looks to be working well. 👍  from me.

**Pull request merged**